File: lecture_notes/08 collections/8-map.cpp
#include <iostream>
#include <map>
#include <string>

using namespace std;


void print_map(const map<string, int>& m) {
    cout << "Map: { ";
    for (const auto& kv : m) {
        cout << kv.first << " = " << kv.second << " ";
    }
    cout << "}" << endl;
}

int main() {
    map<string, int> ages;

    ages.insert({"Vasia Pupkin", 19});
    ages.insert({"Marcus Aurelius", 1860});
    print_map(ages);

    ages["Vasia Pupkin"] = 20;
    print_map(ages);

    ages.erase("Vasia Pupkin");

    cout << boolalpha;
    cout << "Marcus Aurelius: " << (ages.find("Marcus Aurelius") != ages.end()) << endl;
    cout << "Vasia Pupkin: " << (ages.find("Vasia Pupkin") != ages.end()) << endl;

    ages.clear();
    print_map(ages);

    return 0;
}

File: 8-map.py
# 'use strict';
# const ages = new Map();
# ages.set('Vasia Pupkin', 19);
# ages.set('Marcus Aurelius', 1860);
# console.log({ ages });
# ages.set('Vasia Pupkin', 20);
# console.log({ ages });
# ages.delete('Vasia Pupkin');
# console.log({ ages });
# console.log({
#   'Vasia Pupkin': ages.has('Vasia Pupkin'),
#   'Marcus Aurelius': ages.has('Marcus Aurelius'),
# });
# ages.clear();
# console.log({ ages });

# ---------------------------------------------------------------

_contains = lambda x: ages.__contains__(x)


ages = {'Vasia Pupkin': 19, 'Marcus Aurelius': 1860}
print('original:', ages)
ages['Vasia Pupkin'] = 20
print('changed age:', ages)
ages.pop('Vasia Pupkin')
print('delete:', ages)

ages_keys = ['Vasia Pupkin', 'Marcus Aurelius']
ages_values = map(_contains, ages_keys)
ages_dict = dict(zip(ages_keys, ages_values))
print('check:', ages_keys, '>>', ages_dict)

ages.clear()
print('clear:', ages)

